var/spack/repos/builtin/packages/hypre/package.py

```diff
@@ -74,6 +74,7 @@ class Hypre(AutotoolsPackage, CudaPackage, ROCmPackage):
     variant(
         "superlu-dist", default=False, description="Activates support for SuperLU_Dist library"
     )
+    variant("lapack", default=False, description="Use external blas/lapack")
```

Comment:
Does this mean otherwise `hypre` will build its own internal `lapack` / `blas`?

Reply:
Then I'm not sure this is a wanted change, what would be the benefit? In principle, the internal libraries might clash with other `lapack` providers in the DAG.

Reply:
The benefit is shortening the default list of dependencies and improving build time (if blas/lapack are not readily available). 

How can I test if a clash would occur?

Reply:
Hmm, how does that improve the build time? Wouldn't you build `lapack` internally?

Reply:
We actually have a subset of lapack/blas only with the functions we need, so it's faster. Also, building openblas (if blas/lapack are not given) often requires building other libraries

Reply:
I guess my general point is:
- If you have your own internal symbols, and this package is not used as a library, then this is fine
- If you're trying to build a partial lapack/blas with the same symbols, that's bad and should be avoided

Basically, don't hide real dependencies from Spack, cause otherwise it cannot manage them properly.

Reply:
PS: we don't export lapack/blas symbols, instead we prefix them

```
$ nm /home/victor/projects/spack-fork/opt/spack/install-ubuntu24.04-zen4-gcc-14.0.1/hypre-2.32.0/lib/libHYPRE.so | grep 'dasum'
000000000033ab70 T hypre_dasum

```

Reply:
> Would it make sense to have this option, but make it true by default?

I think the default should be "external" blas in a package manager like spack - where invariably hypre is built with other pkgs..

Reply:
Done in [846b30a](https://github.com/spack/spack/pull/47780/commits/846b30a7ee0e58eff53236b1a712f0936819d49f)